server/router/src/main/java/org/infinispan/server/router/router/impl/singleport/RespDetector.java

```diff
@@ -0,0 +1,67 @@
+package org.infinispan.server.router.router.impl.singleport;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import org.infinispan.server.core.transport.AccessControlFilter;
+import org.infinispan.server.resp.RespServer;
+import org.infinispan.server.router.logging.RouterLogger;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerAdapter;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.ByteToMessageDecoder;
+
+/**
+ * Detect RESP connections
+ */
+public class RespDetector extends ByteToMessageDecoder {
+   public static final String NAME = "resp-detector";
+   private final RespServer respServer;
+   private static final Pattern RESP3_HANDSHAKE = Pattern.compile("^(?s)\\*[1-9]\r\n\\$[1-9]\r\n(HELLO|AUTH)\r\n.*");
+
+   public RespDetector(RespServer respServer) {
+      this.respServer = respServer;
+   }
+
+   @Override
+   protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) {
+      // We need to only see the RESP HELLO message
+      if (in.readableBytes() < 5) {
+         // noop, wait for further reads
+         return;
+      }
+      int i = in.readerIndex();
+      if (in.getByte(i) == 42) {
+         // The Redis CLI sends commands as array. Accept it only if it is a HELLO or AUTH command
+         CharSequence handshake = in.getCharSequence(i, in.readableBytes(), StandardCharsets.US_ASCII);
+         if (RESP3_HANDSHAKE.matcher(handshake).matches()) {
+            installRespHandler(ctx);
+         } else {
+            out.add("-ERR Only RESP3 supported");
+         }
+      } else if (in.getCharSequence(i, 5, StandardCharsets.US_ASCII).equals("HELLO")) {
```

Comment:
Either or the line https://github.com/infinispan/infinispan/pull/9837/files#diff-d33f51c567f078daa8e2d5cbc94e3128931a8f9884cd387eca3a6518db355cc9R43 need to change. Either we support RESP2 or not :)

Reply:
I thought we only support RESP3. Also, I don't see what needs to be changed.

Reply:
This line is for RESP2 as it doesn't have the * first.

Reply:
I meant to tag @tristantarrant originally as I wasn't sure what he had decided before.

Reply:
We could support RESP2 :)

Reply:
So, what's the plan here? Are we adding RESP2 support in this PR? I guess the answer is no :) 
In that case, you don't need to install the handler and just do `out.add("-ERR Only RESP3 supported\r\n");`

Reply:
The lettuce parser supports both afaik, so it should work with RESP2.